front/rule.common.php

```diff
@@ -80,10 +80,10 @@
     $rulecollection->checkGlobal(UPDATE);
 
    // Current time
-    $start = explode(" ", microtime());
-    $start = $start[0] + $start[1];
+    $start = microtime(true);
 
    // Limit computed from current time
+    /** @var number $max */
     $max = get_cfg_var("max_execution_time");
```

Comment:
https://www.php.net/manual/en/function.get-cfg-var.php says that it returns `string|array|false`, so `$max` can't be typed as a "number" (did you meant `int` btw ?) without some type casting.

```suggestion
    $max = (int) get_cfg_var("max_execution_time");
```